### PR TITLE
Attach Role to Group and GGC Snap V1.11 Fixes

### PR DESCRIPTION
**Pull request by gvatreya**
**Details**
- Attach Role to Group, handling ExistsAlready scenario
- Add DefaultConfig to set the Execution parameters
  - IsolationMode = NoContainer
  - Uid and Gid = 584788
- Changes for deploying to UbuntuCore where GGC runs as snap
  following the steps outlined in
https://docs.aws.amazon.com/greengrass/v1/developerguide/install-ggc.html

### CODE REVIEW

File: gg_group_setup/group.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def create_and_attach_iam_role(self):
         gg_client = boto3.client('greengrass', region_name=self.region)
         role_name = '{0}_service_role'.format(self.type_name)
         aws_lambda_ro_access_arn = \
-            "arn:aws:iam::aws:policy/AWSLambdaReadOnlyAccess"
+            "arn:aws:iam::aws:policy/AWSLambda_ReadOnlyAccess"
```
Comment (gvatreya) [inline gg_group_setup/group.py:266]: The policy is deprecated - had to use the new one.

File: gg_group_setup/group.py [hunk 2]
```diff
@@ -293,45 +293,85 @@ def create_and_attach_iam_role(self):
...
+            logging.info("associate_service_role_to_account completed", json.dumps(resp))
+        except ClientError as ce:
+            logging.debug("Caught Error", exc_info=True, stack_info=True)
+            if ce.response['Error']['Code'] == 'ResourceAlreadyExistsException':
+                logging.warning("[put_policy] {0}".format(ce.response['Error']['Message']))
+            else:
+                logging.error("[put_policy] {0}".format(ce.response['Error']['Message']))
+            # role already exists return nothing, assuming previous success
 
+        try:
```
Comment (gvatreya) [inline gg_group_setup/group.py:362]: New code to attach role to group

File: gg_group_setup/group.py [hunk 3]
```diff
@@ -293,45 +293,85 @@ def create_and_attach_iam_role(self):
                 }
             ]
         }
+
```
Comment (gvatreya) [inline gg_group_setup/group.py:296]: Separated our the error handlings for each sub step

File: gg_group_setup/cmd.py [hunk 4]
```diff
@@ -278,20 +280,33 @@ def _create_function_definition(gg_client, group_type, config, region=None):
...
         # if we found one or more configured functions, create a func definition
+        # Since Ubuntu Core uses snap and GGC for snap subsystem, supports
+        # only noncontainerized lambda functions and must be run with uid/gid 584788
+        # Refer the section Run AWS IoT Greengrass in a snap > Deploying a Lambda function > Step 4e
+        # https://docs.aws.amazon.com/greengrass/v1/developerguide/install-ggc.html
         if len(func_definition) > 0:
             ll = gg_client.create_function_definition(
                 Name="{0}_func_def".format(group_type.type_name)
             )
             lmbv = gg_client.create_function_definition_version(
```
Comment (gvatreya) [inline gg_group_setup/cmd.py:300]: Refer https://docs.aws.amazon.com/greengrass/v1/developerguide/install-ggc.html  : Deploy Lambda function Step 4.e.
Screenshot 2021-04-05 at 5 54 49 PM
Screenshot 2021-04-05 at 5 54 43 PM

**Review by dostiharise — APPROVED** [review]
Validate your understanding on IAM Roles.

**Comment by gvatreya** [comment]
> Validate your understanding on IAM Roles.

@dostiharise I and @rahulsrp verified that previously, the script was not attaching the role to the group.
I am merging this now. Will create a new PR if you need any changes.

**Pull request merged**